### PR TITLE
ninja: Remove conan_version import

### PR DESCRIPTION
**Pull request by jwillikers**
This breaks older versions of Conan, unfortunately.

Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by System-Arch** [comment]
This change will cause failures with Conan 2 prior to Beta 5 being released (see https://github.com/conan-io/conan/issues/12334)

**Comment by SpaceIm** [comment]
> This breaks older versions of Conan, unfortunately.

So why this PR? It doesn't improve support conan v1 client and breaks current conan v2 client.

**Comment by jwillikers** [comment]
> 

Here's the error:

```
ERROR: ninja/1.11.1: Cannot load recipe.
Error loading conanfile at '/home/user/.conan/data/ninja/1.11.1/_/_/export/conanfile.py': Unable to load conanfile in /home/user/.conan/data/ninja/1.11.1/_/_/export/conanfile.py
  File "/usr/lib/python3.8/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/user/.conan/data/ninja/1.11.1/_/_/export/conanfile.py", line 1, in <module>
    from conan import ConanFile, conan_version
ImportError: cannot import name 'conan_version' from 'conan' (/usr/local/lib/python3.8/dist-packages/conan/__init__.py)
```

**Comment by SpaceIm** [comment]
It requires conan 1.52.0, that's all.

**Comment by SSE4** [comment]
@uilianries is it possible to use something like:
```
try:
    from conan import conan_version
except ImportError
    conan_version = 0
```

**Comment by jwillikers** [comment]
> It requires conan 1.52.0, that's all.

It specifies `required_conan_version = ">=1.52.0"`, but still caused this error. Why is that?

**Comment by SpaceIm** [comment]
> > It requires conan 1.52.0, that's all.
> 
> It specifies `required_conan_version = ">=1.52.0"`, but still caused this error. Why is that?

Because conan clients before 1.52.0 were not able to check `required_conan_version` before interpreting python imports  (https://github.com/conan-io/conan/pull/11908).

File: recipes/ninja/all/conanfile.py [hunk 1]
```diff
@@ -49,6 +48,5 @@ def package_info(self):
         self.conf_info.define("tools.cmake.cmaketoolchain:generator", "Ninja")
 
         # TODO: to remove in conan v2
-        if Version(conan_version).major < 2:
-            self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
-            self.env_info.CONAN_CMAKE_GENERATOR = "Ninja"
+        self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
+        self.env_info.CONAN_CMAKE_GENERATOR = "Ninja"
```
Comment (prince-chrismc) [inline recipes/ninja/all/conanfile.py:52]: Should this be added to build env for the new generators?

**Comment by prince-chrismc** [comment]
> I don't understand this relentlessness against conan_version.

We need to make sure the client is up to par. The easiest way to find problems is to let them happen.

We want to measure how successful the v2 pipeline will be and how close the migration process is to completion.

The goal is recipes should work in both... if we need to check the version... obviously something is wrong. So let's not hide the problems.

There's too many PRs for us to manually review them so it need to error out so we can track what's going on.


Hopefully that clears it up :)

File: recipes/ninja/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/ninja/all/conanfile.py:52]: I would say more, it should configure `tools.cmake.cmaketoolchain:generator`

https://docs.conan.io/en/latest/reference/config_files/global_conf.html

**Comment by uilianries** [comment]
> @uilianries is it possible to use something like:
> 
> ```
> try:
>     from conan import conan_version
> except ImportError
>     conan_version = 0
> ```

I would prefer:

```
try:
    from conan import conan_version
except ImportError
    from conans import __version__ as conan_version
```

Which is backward compatible.

File: recipes/ninja/all/conanfile.py (hunk 1 above)
Comment (jwillikers) [reply recipes/ninja/all/conanfile.py:52]: The recipe already defines the conf value for the generator. Should I additionally add it to build_env path?

**Comment by jwillikers** [comment]
@uilianries Thanks, that change is perfect!

File: recipes/ninja/all/conanfile.py [hunk 2]
```diff
@@ -1,6 +1,11 @@
 from conan import ConanFile
 from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
 from conan.tools.files import copy, get
+from conan.tools.scm import Version
+try:
+    from conan import conan_version
+except ImportError
+    from conans import __version__ as conan_version
```
Comment (SpaceIm) [inline recipes/ninja/all/conanfile.py:8]: https://github.com/conan-io/conan-center-index/issues/12124#issuecomment-1210327220

Comment (System-Arch) [reply recipes/ninja/all/conanfile.py:8]: It seems like many of these issues result from trying to run Conan 2.0 recipes through Conan 1.x. Why did CCI not elect to simply create a parallel branch for 2.0 work analogous to how the Conan project divides its efforts.  Adding 1.x workarounds like this significantly detract from the simplicity and elegance of pure 2.0 recipes. There appear to be quite a few issues like this where there is no single solution that works for both 1,x and 2.0.

Comment (uilianries) [reply recipes/ninja/all/conanfile.py:8]: @System-Arch All recipes will be migrated and CCI will start to adopt Conan 2.x more and more. Please, check the roadmap: https://github.com/conan-io/conan-center-index/blob/master/docs/v2_roadmap.md

Comment (jwillikers) [reply recipes/ninja/all/conanfile.py:8]: Thanks for the links and opening that issue @SpaceIm. Everyone in agreement on me closing this PR then?

Comment (uilianries) [reply recipes/ninja/all/conanfile.py:8]: @jwillikers yes please.

**Pull request closed without merging**